cmake/modules/BuildBoost.cmake

```diff
@@ -130,7 +138,19 @@ function(do_build_boost version)
   # edit the user-config.jam so, so b2 will be able to use the specified
   # toolset
   file(WRITE ${source_dir}/user-config.jam
-    "using ${toolset} : : ${CMAKE_CXX_COMPILER} ;")
+    "using ${toolset}"
+    " : "
+    " : ${CMAKE_CXX_COMPILER}"
+    " ;\n")
+  if(with_python GREATER -1)
+    file(APPEND ${source_dir}/user-config.jam
+      "using python"
+      " : ${PYTHON_VERSION_STRING}"
+      " : ${PYTHON_EXECUTABLE}"
+      " : ${PYTHON_INCLUDE_DIRS}"
+      " : ${PYTHON_LIBRARIES}"
```

Comment:
It seems that:

1. in the py2/py3 environment, these variables are populated by FindPython{Interp,Libs}.cmake and they point to Python 2.7
2. in the py3-only environment, `FindPython{Interp,Libs}.cmake` is not run. Instead, `FindPython3{Interp,Libs}.cmake` is run and does not populate these variables. Instead of `PYTHON_VERSION_STRING`, for example, it populates `PYTHON3_VERSION_STRING`.

Reply:
@tchaikov Please ignore the above comment - I found that `FindPython{Interp,Libs}.cmake` are properly populated in the py3-only env.